runtime/vm/jnimisc.cpp

```diff
@@ -520,6 +520,7 @@ newStringUTF(JNIEnv *env, const char *bytes)
 				gpCheckSetNativeOutOfMemoryError(currentThread, 0, 0);
 				goto done;
 			}
+			data[length*2-1] = '\0';
```

Comment:
This can potentially corrupt data if string needed exactly `length*2` bytes when converted.  I think you still need to over-allocate by 1 for the null